public/js_v2.0.1_f1/chatlog_functions.js

```diff
@@ -232,6 +232,11 @@ function loadMessagesOnGUI(messages) {
     threads.forEach(thread => {
         checkThreadUnreadMessages(thread);
     });
+
+    // set the model in the current conversation
+    // as the one used in the last message
+    let lastMessage = messages.at(-1);
```

Comment:
Can this ever fail, i.e. what happens if `messages.length := 0`? Maybe just safeguard the line below? 

```
if (lastMessage && lastMessage.model) { setModel(lastMessage.model); }